### DISCUSSION TITLE
RAG Model for Single note; Where is it?

### DISCUSSION THREAD

**Discussion by FamedBear16** [logancyang/obsidian-copilot#772] [General]
Hi,
In the past I could use the RAG model on a single mode. it was working very well for me. 
Can you please tell me how I can use it now? I do not see the option any longer.
I'd rather not use the Vault mode, because it is indexing all my vault, while I just need all the files in one directory with the exclusion of excalidraw drawings.

Many thanks.

**Comment by logancyang** [comment]
Single note RAG isn't available anymore. Perhaps an inclusion filter for Vault QA is needed then you can just index one note or a subset of your vault instead of using exclusion.

**Reply by FamedBear16** [reply]
HI @logancyang,

It would be nice to have inclusion filters.
The removal of the old options, single RAG note, is not working well for me.
I have a ZK vault, with many small notes. Therefore, the Vault QA is not going to work well for me, since it is limited to max 10 notes.
I raised this issue in the past while you were working at it. 
It would be better to have a limit in terms of numbers of tokens submitted, rather than the number of notes. 

My work around was to write a single file with all my filtered Vault notes and query this single note instead. It was giving much better result the the Vault QA in the past.

I would appreciate either  on Inclusion filter (best solution) or a Single Note RAG options.

I have another question. If I exclude a folder ex: "References" are all the subfolder excluded as well?
if not, it would be nice to have a way to exclude all the subfolder and include back some of them

Regards,

**Reply by logancyang** [reply]
Yes it should exclude all subfolders, if it's not please file a bug!

I will introduce the Inclusion filter in the coming release this week! Sorry for the trouble. @FamedBear16

**Reply by FamedBear16** [reply]
Hi @logancyang ,
Thanks for your help

**Reply by logancyang** [reply]
@FamedBear16 Inclusion filter is live in v2.7.1, please give it a try!
